pmd-test-schema/src/main/java/net/sourceforge/pmd/test/schema/BaseTestParserImpl.java

```diff
@@ -204,7 +206,10 @@ private String getTestCode(Element testCode, Map<String, Element> fragments, Set
             }
             usedFragments.add(id.getValue());
             code = parseTextNodeNoTrim(fragment);
-            code = code.trim(); // todo replace with trimIndent in PMD 7
+            // first trim empty lines at beginning/end
+            code = code.trim();
+            // then trim any indentation
+            code = StringUtil.trimIndent(Chars.wrap(code)).toString();
```

Comment:
this won't work…

given a multiline value such as:

```
   this is
   a sample
   text
```

Bu doing `trim()` first you get:


```
this is
   a sample
   text
```

And this will cause `StringUtil.trimIndent` to believe there is no shared indent among all lines.

To actually remove empty lines at the beginning / end without affecting indents you should use `Chars.trimBlankLines`

Reply:
True, good catch :)
I honestly didn't test that this does what it is supposed to do, just blindly implementing the TODO...

In the rule tests, we only check the line numbers and not the columns - that's how I figured, that we need the extra trim to get rid of the blank line at the beginngin.

I'll have look at this....

Reply:
I believe just doing

```suggestion
            // first trim empty lines at beginning/end, then trim any indentation
            code = StringUtil.trimIndent(Chars.wrap(code).trimBlankLines()).toString();
```

should work as expected…